docs/specifications/tests/Zone-TP/zone01.md

```diff
@@ -3,43 +3,209 @@
+
+3. Obtain the set of name server IP addresses using [Method4] and [Method5] ("Name Server IP").
+
+4. For each name server IP address in *Name Server IP* do:
+   1. Send *SOA Query* to the name server and collect the response.
+   2. Go to next name server IP address if at least one of the following criteria is met:
+      1. There is no DNS response.
+      2. [RCODE Name] of the response is not "NoError".
+      3. The AA flag is not set in the response.
+      4. There is no SOA record with owner name matching the query.
+   3. From the DNS response, extract the name server name from the MNAME field:
+      1. If the name is "localhost", then add name server IP address to the *MNAME Is Localhost* set and go to next name server.
+      2. Else if the name is ".", then add name server IP address to the *MNAME Is Dot* set and go to next name server.
+      3. Else, add name server name to the *MNAME Nameservers* set.
+   4. From the DNS response, add the SERIAL field and name server IP address to the *SERIAL Nameservers* set.
+   5. Go to next name server.
+
+5. If the set *MNAME Is Localhost* is non-empty, then output *[Z01_MNAME_IS_LOCALHOST]* with name servers IP address(es).
+
+6. If the set *MNAME Is Dot* is non-empty, then output *[Z01_MNAME_IS_DOT]* with name servers IP address(es).
+
+7. If the set *MNAME Nameservers* is empty, then terminate these procedures.
+
+8. Else, for each name server name in *MNAME Nameservers* do:
+   1. Do a [DNS Lookup] of name server name (A and AAAA) and add the name server IP address(es) found to the *MNAME Nameservers* set.
+   2. If at least one IP address from the previous step was found, then:
+      1. For each name server IP for the name server name do: 
+         1. Send *SOA Query* to the name server and collect the response.
+            1. If there is a DNS response with [RCODE Name] "NoError", then:
+               1. If the AA flag is not set, then output *[Z01_MNAME_NOT_AUTHORITATIVE]* with name server name and IP address.
+               2. Else, add the SERIAL field to the *MNAME Nameservers* set.
+            2. Else if there is a DNS response with [RCODE Name] other than "NoError", then output *[Z01_MNAME_UNEXPECTED_RCODE]* with [RCODE Name], name server name and IP address.
+            3. Else, output *[Z01_MNAME_NO_RESPONSE]* with name server name and IP address.
+         2. Go to next name server IP.
+   3. Else, output *[Z01_MNAME_NOT_RESOLVE]* with name server name.
+   4. Go to next name server name.
+
+9. If there is no SERIAL in the set *MNAME Nameservers*, then terminate these procedures.
+
+10. For each SERIAL in *SERIAL Nameservers* do:
+   1. For each SERIAL in *MNAME Nameservers* do:
+      1. Compare both SERIAL values (using the arithmetic in [RFC1982]).
+         1. If the one from *SERIAL Nameservers* is not higher, then add name server IP address to the *MNAME Is Master* set.
+         2. Else, add name server IP address to the *MNAME Not Master* set.
```

Comment:
For programming this is fine, but it is quite complex for manual execution. And what IP address is "name server IP address"?

Reply:
I have updated and reworded this step to make it clearer. Although it may still be a struggle to read, let me know.